packages/govuk-frontend/src/govuk/components/button/button.test.js

```diff
@@ -359,6 +338,45 @@ describe('/components/button', () => {
           message: 'GOV.UK Frontend is not supported in this browser'
         })
       })
+
+      it('throws when $module is not set', async () => {
+        await expect(
+          renderAndInitialise(page, 'button', {
+            params: examples.default,
+            beforeInitialisation() {
+              // Remove the root of the components as a way
+              // for the constructor to receive the wrong type for `$module`
+              document.querySelector('[data-module]').remove()
+            }
+          })
+        ).rejects.toEqual({
+          name: 'MissingElementError',
+          message: 'Button: $module not found'
+        })
+      })
+
+      it('throws when receiving the wrong type for $module', async () => {
+        await expect(
+          renderAndInitialise(page, 'button', {
+            params: examples.default,
+            beforeInitialisation() {
+              // Remove the root of the components for replacement
+              document.querySelector('[data-module]').remove()
+
+              // And replace with an `<svg>` element which doesn't have the right type
+              const $svg = document.createElementNS(
+                'http://www.w3.org/2000/svg',
+                'svg'
+              )
+              $svg.setAttribute('data-module', 'button')
```

Comment:
Probably want to lock this down but `[data-module]` works with any value:

https://github.com/alphagov/govuk-frontend/blob/0443c375979978fb2321267774151b4806a81fb5/shared/helpers/puppeteer.js#L115

But shall we stay consistent and always follow ~`button`~ `govuk-button` which `initAll()` uses?

https://github.com/alphagov/govuk-frontend/blob/0443c375979978fb2321267774151b4806a81fb5/packages/govuk-frontend/src/govuk/all.mjs#L54-L56

Reply:
I like the idea of referring to `Component.moduleName`, but I think it'd make sense to do that as part of a work that'd make our test helpers (`goToComponent`, `renderAndInitialise`,...) all rely on the component's class to get that kind of information (would ensure things are systemised from one component to another as well). If that's fine with you, I'd be keen to track that as a separate issue from this PR 😊

Reply:
Yeah sounds good. It might catch us out with nested components one day 👍

Thanks Romaric